cache/network/api.go

```diff
@@ -143,14 +139,56 @@ func (c apiClient) uploadArchive(archivePath, uploadMethod, uploadURL string, he
 	c.logger.Debugf("Response dump: %s", string(dump))
 
 	if resp.StatusCode != http.StatusOK {
-		return unwrapError(resp)
+		return "", unwrapError(resp)
+	}
+
+	etag := resp.Header.Get("ETag")
```

Comment:
Is it guaranteed that this header exists on the response?

Reply:
On S3 this is a must, on Google this won't be there AFAIK.

Reply:
This Get method returns an empty string if there wasn't a header so this should work